modules/internal/ChapelArray.chpl

```diff
@@ -2358,15 +2360,25 @@ module ChapelArray {
 
     */
     pragma "fn returns aliasing array"
-    proc reindex(d: domain)
-      where isRectangularDom(this.domain) && isRectangularDom(d)
+    inline proc reindex(newDomain: domain)
+      where isRectangularDom(this.domain) && isRectangularDom(newDomain)
+    return reindex((...newDomain.dims()));
+
+    pragma "no doc"
+    pragma "fn returns aliasing array"
+    proc reindex(newDims...)
```

Comment:
I think this function signature to be:
`proc reindex(newDims: range... rank)`

And then another more generic function could be added with "no doc" to handle the compilerErrors.